client/lib/explat/internals/fetch-experiment-assignment.ts

```diff
@@ -0,0 +1,27 @@
+/**
+ * WordPress dependencies
+ */
+import wpcom from 'calypso/lib/wp';
+
+// Using typescript to ensure we are being safe in SSR
```

Comment:
What does this do?

Also, looking at #50261, do we need this in every internal file? Can we get the same effect by moving it to the index file?

Reply:
> What does this do?

This expands `window`'s type to include undefined, this means typescript is helping us be SSR safe.

> Also, looking at #50261, do we need this in every internal file? Can we get the same effect by moving it to the index file?

I can't seem to get it to work by importing, and if it did work somehow we would still need a line in each file.

Reply:
I still don't understand. How does it help us to have an unused `window` declaration in this file and in other files where `window` is not referenced?

Looking at [other `declare const window` calls](https://github.com/Automattic/wp-calypso/search?q=%22declare+const+window%22), they're always used in files where `window` is referenced.

Reply:
It is a preventative measure: any reference to window in any of these files could cause a crash in SSR conditions, by adding we get a squiggly red line whenever `window` is used without type-guarding undefined away.

So even though there isn't a reference to `window` in these files, if it gets introduced, this will let the future developer know that something is up with it.

Reply:
Ah, thanks for clarifying! How about we change the phrasing here an elsewhere to something like this:

```suggestion
// SSR safety: Fail TypeScript compilation if `window` is used without an explicit undefined check
```

Reply:
Fixing this in #50474